[Fun Leon] iP

niekis opened the pull request:
### Fideline

> "I'm gonna put some dirt in your eye" - Peter Parker (a task he plans to complete)

Keeps track of tasks for you! With 3 new types
- `Todo` (simple)
- `Deadline` (has a due date)
- `Event` (has a start and end time)

Why should you use Fideline?
1. ~~Clutter~~ Declutter your mind!
2. Remember what tasks you have left!
3. Stores your tasks that can be view in **future sessions!**

running it will feel like:
```
public class Main{
    public static void main(String[] args) {
        Fideline.makeLifeEasier();
        Fideline.thisIsJustAJoke();
    }
}
```

Now you can keep track of tasks and their *completeness status!* 😮  
- [ ] task 1
- [ ] task 2

[watch out](https://github.com/nus-cs2103-AY2223S2/ip/pull/332#pullrequestreview-1279001342)

Commit:
Add Gradle support

niekis committed:
Level-1
add Fideline.java
edited Duke.java to employ Fideline bot

niekis committed:
Level-2
added list function
tweaked some responses

niekis committed:
Level-3
added ability to mark/unmark tasks
added Task.java, TaskManager.java

niekis committed:
Level-4
Task.java change to abstract class with 3 new child classes: Todo.java, Deadline.java, Event.java
User is now able to created 3 types of tasks

niekis committed:
Level-4
Task.java change to abstract class with 3 new child classes: Todo.java, Deadline.java, Event.java
User is now able to created 3 types of tasks
This commit was supposed to be part of the previous one

niekis committed:
added first test case (creation of tasks + marking/unmarking functionality)

niekis committed:
Level-5
currently handling
syntax errors for deadline, event command
wrong type of argument error for mark and unmark command (not int)
no argument error for todo, deadline, event, mark, unmark command
invalid command error

added InputParser.java to handle and parse user input
added custom InputError.java that is to be throw when user input is invalid

niekis committed:
Level-5
currently handling
syntax errors for deadline, event command
wrong type of argument error for mark and unmark command (not int)
no argument error for todo, deadline, event, mark, unmark command
invalid command error

added InputParser.java to handle and parse user input
added custom InputError.java that is to be throw when user input is invalid
*should have been part of previous commit. Forgot to stage changes*

niekis committed:
Level-6
added deletion command to delete existing tasks

niekis committed:
Save task data automatically

Task data is lost after program ends.

Let's store all task data locally.

Task data can be accessed during later sessions.

niekis committed:
Update deadline parser to recognize dates

Deadline task's deadlines are stored exclusively as strings.

Program is unable to make sense of the timing of deadlines.

Let's update the parser to detect deadline tasks created with
a deadline following the format "dd/mm/yyyy" and store deadlines
as dates.

Dates can be expressed more intuitively by the system. Features that
compare the deadlines of tasks can be added in the future.

niekis committed:
Merge branch 'branch-Level-7'

niekis committed:
Merge branch 'branch-Level-8'

niekis committed:
Deadline, Event, Fideline class: fix message spacing

niekis committed:
Make the code more OOP

Fideline and Inputparser files are densely packed and hard to read.

Code is difficult to read and edit.

Let's organize the code in a more readable fashion, following OOP
closely. More classes are created for better organization.

Code becomes more organized with greater standardization.

niekis committed:
Fix bugs relating to saving and reading saved data

niekis committed:
Merge remote-tracking branch 'origin/add-gradle-support'

junianob commented on src/main/java/fideline/user/Ui.java:
```diff
@@ -0,0 +1,97 @@
+     * Bids the user goodbye.
+     */
+    public void goodbye() {
+
```
random newline here, forgot to delete maybe?

junianob commented on src/main/java/fideline/user/Parser.java:
```diff
@@ -0,0 +1,119 @@
+package fideline.user;
+
+import fideline.exception.FidelineException;
+import fideline.exception.EmptyParamException;
+import fideline.exception.InvalidArgumentException;
+import fideline.exception.UnknownCommandException;
+import fideline.exception.WrongFormatException;
+
+import fideline.execution.Command;
+import fideline.execution.ExitCommand;
+import fideline.execution.ListCommand;
+import fideline.execution.CreateTodoCommand;
+import fideline.execution.CreateDeadlineCommand;
+import fideline.execution.CreateEventCommand;
+import fideline.execution.MarkCommand;
+import fideline.execution.UnmarkCommand;
+import fideline.execution.DeleteCommand;
+
+
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.time.LocalDate;
```
Good practice of not using wildcard imports

junianob commented on src/main/java/fideline/task/TaskManager.java:
```diff
@@ -0,0 +1,107 @@
+        return taskNumber <= taskList.size();
+    }
+
+    // get the name of the task at a given index
```
description is not exactly accurate, getTaskString gets the task name as well as the task type and done status (not sure if intended, this is a minor issue)

junianob commented on src/main/java/fideline/Fideline.java:
```diff
@@ -0,0 +1,54 @@
+package fideline;
+
+import fideline.exception.*;
```
Forgot about no wildcard imports here

junianob reviewed (commented):
LGTM. Generally code is formatted neatly and follows the style guide, with some minor nitpicks here and there that can be easily fixed. 👍

niekis committed:
Add Gradle and JUnit

Let's use Gradle and JUnit to automate and aid in the building of this
project.

niekis committed:
Add javadoc to a few classes

niekis committed:
Tweak the code to comply with coding standards

niekis committed:
Add find command allowing user to find tasks using keywords.

Let's add a find command that can provide a list of all existing tasks
that contain a specific keyword in their description.

niekis committed:
Merge branch 'branch-A-CodingStandard'

niekis committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/fideline/task/TaskManager.java
#	src/main/java/fideline/user/Parser.java

niekis committed:
Merge branch 'branch-A-JavaDoc'

niekis committed:
Improve code's style using checkstyle

niekis committed:
Add GUI for Fideline

Fideline employs a text-based UI.

Expression is limited and it is not very user-friendly.

Let's implement GUI for Fideline.

Fideline's user-friendliness can be enhanced through this format.

niekis committed:
Add some documentation

niekis committed:
Fix dialog box cutting off long text

niekis committed:
Merge branch 'branch-Level-10'

niekis committed:
Merge branch 'branch-A-Checkstyle'

niekis committed:
Improve the overall code quality

Code is hard to read in some areas and the quality can be improved in
most files.

Code is difficult to read and edit.

Let's improve the quality of the code, paying close attention to
SLAP, guard clauses and avoiding arrowheads.

Code becomes more organized with greater standardization.

niekis committed:
Add assertions

Let's add assertions to improve code security while edits are made.

niekis committed:
Merge pull request #2 from niekis/branch-A-CodeQuality

Improve the overall code quality

niekis committed:
Merge branch 'master' into branch-A-Assertions

niekis committed:
Merge pull request #3 from niekis/branch-A-Assertions

Add assertions

niekis committed:
Check if new task proposed already exists in the task list

Duplicate tasks are not accounted for.

User can accidentally create the same task twice.

Let's prevent the user from creating duplicate tasks, throwing an
exception if the new task already exists in the task list.

(C-DetectDuplicates)

niekis committed:
Merge branch 'branch-C-DetectDuplicates'

niekis committed:
Add user guide to the readme.md

niekis committed:
Update readme

Let's include quick start portion to the user guide so that new users
will know how to start the program using a .jar file

niekis committed:
Prepare for release

niekis committed:
Fix mistakes in UG
